### ISSUE TITLE
Missing Code-Assists On Models

### ISSUE THREAD

**Issue by saifulss** [barryvdh/laravel-ide-helper#453]
Creating this thread to report code-assists that were available when using v2.2 with Laravel 5.3.30, but aren't anymore when using same v2.3 with Laravel 5.4.9.

* Illiustrations use a model 'Project'.

**The following code are missing code-assists:**
$project = App\Project::create(['name' => 'Some Project']);    //Model::create()
$project = App\Project::find(1);    //Model::find()

Will post more as I find them.

Are these known issues? Am I missing some configuration step?

**Comment by barryvdh** [comment]
And 2.3.1 ? Are you extending Eloquent or Model?

**Comment by saifulss** [comment]
Just tried again. Fresh Laravel 5.4.12, Helper 2.3.1.

Project now extends Eloquent, not Model - I usually use 'php artisan make:model Project' which puts it as extends Model.

Whether extending Model or Eloquent, I get the same results - Project::find() and Project::create() doesn't have code-assists.

Let me know how else I can assist. This package is easily the most useful Laravel package for me and I'd like to contribute where I can.

**Comment by briedis** [comment]
Yeah, just updated to 2.3.1 and all static model calls are now highlighted :( Reverting back...
I extend Model, but have `@mixin \Eloquent` commont at the top. Tried extending Eloquent - same.

I see that 2.3.1 removes Eloquent facade, so this kinda makes sense, even though I have never had issues with breaking autocomplete while using this package.

Running PhpStorm EAP 2017.1

**Comment by barryvdh** [comment]
Can you try with https://github.com/barryvdh/laravel-ide-helper/releases/tag/v2.3.2 ?

**Comment by briedis** [comment]
@barryvdh yes, this solves the problem. Phpstorm inspec does not detect any problems, auto-complete works fine. EAP 2017.1

**Comment by saifulss** [comment]
I can also confirm that Helper 2.3.2 on Laravel 5.4.12 is now showing User::find() and User::create().

That solves that issue. Thanks Barry!

Further to that, I found that User::find(1) returns a 'mixed' instead of an instance of the User model class. Which then breaks code-assist on all else coming after the find().

E.g. User::find(1)->whereName('Bob Loblaw')->first();

Note that I've already done:
```
php artisan ide-helper:generate
php artisan ide-helper:models -RW
```

I remember this to work well back in Laravel 5.3.* using Helper 2.2.*.

Should I start a fresh issue?

**Comment by jonnywilliamson** [comment]
https://github.com/laravel/framework/commit/3be29a0851e44a038a020fea0804dce03f01863b

No I think it;s more fundamental. We need taylor to revert or change the doctype return here.

**Comment by briedis** [comment]
@jonnywilliamson haha, it would be nice if he used and IDE instead of a text editor. Laravel's phpdocs are kinda driving me mad sometimes.....

**Comment by jonnywilliamson** [comment]
@briedis Yeah, the trouble is when you write every line of your own software, you remember how things work.

For the rest of us poor mortals, having an IDE is literally the only way to save a LOT of time fixing typos and miswritten method names etc.

**Comment by saifulss** [comment]
Thanks for the reference, @jonnywilliamson. Cleared it up quite a bit.

I guess there's little more to do here for this repo per se. Just gotta hope Taylor returns it back to the way it was.

**Comment by michaeljhopkins** [comment]
I THINK this may be the same issue. I have the following code (some "assumed" lines removed for brevity)

```php

class Thread extends Model {
     public function path() {
        return '/threads/'.$this->id;
    }
}

class ThreadsController extends Controller {
    public function store(Request $request) {
        $thread = Thread::create($validArray);

        // Issue is here. No autocomplete for the path method. 
        // $thread is considered an instance of \I..\D..\E..\Model
        $thread->pa...
    }
}
```

> Let me know how else I can assist. This package is easily the most useful Laravel package for me and I'd like to contribute where I can.

I second this sentiment and absolutely want to help keep this package as great as it has always been

**Comment by michaeljhopkins** [comment]
On further investigation it looks like this is the same issue. 

https://github.com/laravel/framework/commit/95f2e35d88f320824807b555d80272ee49402b4a

After moving the methods from `Model`->`Builder`, the return types were changed from `@return static`->`@return \Illuminate\Database\Eloquent\Model`

**Comment by eidng8** [comment]
I've tried with L5.4.19, helper 2.3.2, with a fresh project. Run `ide-helper:meta`, `ide-helper:generate`, then `ide-helper:model`. Code assist does popup.

But then, at the bottom of `app/Http/Controllers/Auth/RegisterController.php`, the call to `User::create()`, PhpStorm complains "expected app\User, but got \Illuminate\Database\Eloquent\Model" stuff.

After changed doc block to `@return static`, PS doesn't complain any more. But I'm not sure if it would break something else if it's changed to `@return static`

**Comment by mfn** [comment]
@barryvdh the original issue was fixed with 2.3.2 (wow, that must have been a hell of a release!) and the other problems mentioned refer to versions not supported anymore => I vote for closing this issue :}
